examples/cluster/plot_adjusted_for_chance_measures.py

```diff
@@ -3,135 +3,231 @@
+    ("AMI", metrics.adjusted_mutual_info_score),
+]
 
-def uniform_labelings_scores(
-    score_func, n_samples, n_clusters_range, fixed_n_classes=None, n_runs=5, seed=42
-):
-    """Compute score for 2 random uniform cluster labelings.
+# %%
+# First experiment: fixed ground truth labels and growing number of clusters
+# --------------------------------------------------------------------------
+#
+# We first define a function that creates uniformly-distributed random labeling.
 
-    Both random labelings have the same number of clusters for each value
-    possible value in ``n_clusters_range``.
+import numpy as np
 
-    When fixed_n_classes is not None the first labeling is considered a ground
-    truth class assignment with fixed number of classes.
-    """
-    random_labels = np.random.RandomState(seed).randint
-    scores = np.zeros((len(n_clusters_range), n_runs))
+rng = np.random.RandomState(0)
 
-    if fixed_n_classes is not None:
-        labels_a = random_labels(low=0, high=fixed_n_classes, size=n_samples)
 
-    for i, k in enumerate(n_clusters_range):
-        for j in range(n_runs):
-            if fixed_n_classes is None:
-                labels_a = random_labels(low=0, high=k, size=n_samples)
-            labels_b = random_labels(low=0, high=k, size=n_samples)
-            scores[i, j] = score_func(labels_a, labels_b)
-    return scores
+def random_labels(n_samples, n_classes):
+    return rng.randint(low=0, high=n_classes, size=n_samples)
 
 
-def ami_score(U, V):
-    return metrics.adjusted_mutual_info_score(U, V)
+# %%
+# Another function will use the `random_labels` function to create a fixed set
+# of ground truth labels (`labels_a`) distributed in `n_classes` and then score
+# several sets of randomly "predicted" labels (`labels_b`) to assess the
+# variability of a given metric at a given `n_clusters`.
 
 
-score_funcs = [
-    metrics.adjusted_rand_score,
-    metrics.v_measure_score,
-    ami_score,
-    metrics.mutual_info_score,
-]
+def fixed_classes_uniform_labelings_scores(
+    score_func, n_samples, n_clusters_range, n_classes, n_runs=5
+):
+    scores = np.zeros((len(n_clusters_range), n_runs))
+    labels_a = random_labels(n_samples=n_samples, n_classes=n_classes)
 
-# 2 independent random clusterings with equal cluster number
+    for i, n_clusters in enumerate(n_clusters_range):
+        for j in range(n_runs):
+            labels_b = random_labels(n_samples=n_samples, n_classes=n_clusters)
+            scores[i, j] = score_func(labels_a, labels_b)
+    return scores
 
-n_samples = 100
-n_clusters_range = np.linspace(2, n_samples, 10).astype(int)
 
-plt.figure(1)
+# %%
+# In this first example we set the number of clases (true number of clusters) to
+# `n_classes=10`. The number of clusters varies over the values provided by
+# `n_clusters_range`.
+
+import matplotlib.pyplot as plt
+import matplotlib.style as style
 
+n_samples = 1000
+n_classes = 10
+n_clusters_range = np.arange(2, 102, 11)
```

Comment:
Seeing the linspace used in the second experiment, where it wants the maximum `n_clusters` to be 100, I think it's better to use `linspace` here to be consistent. 

```suggestion
n_clusters_range = np.linspace(2, 100, 10).astype(int)
```

Sorry for the suggestion before this that converted it to a arange.